src/core/ddsi/defconfig.c

```diff
@@ -0,0 +1,116 @@
+#include <string.h>
+#include <stdint.h>
+#include <inttypes.h>
+#include "dds/ddsi/ddsi_config.h"
+
+void ddsi_config_init_default (struct ddsi_config *cfg)
+{
+  memset (cfg, 0, sizeof (*cfg));
+  static char *networkRecvAddressStrings_init_[] = {
+    "preferred",
+    NULL
+  };
+  cfg->networkRecvAddressStrings = networkRecvAddressStrings_init_;
+  cfg->externalMaskString = "0.0.0.0";
+  cfg->allowMulticast = UINT32_C (2147483648);
+  cfg->multicast_ttl = INT32_C (32);
+  cfg->transport_selector = INT32_C (1);
+  cfg->enableMulticastLoopback = INT32_C (1);
+  cfg->max_msg_size = UINT32_C (14720);
+  cfg->max_rexmit_msg_size = UINT32_C (1456);
+  cfg->fragment_size = UINT16_C (1344);
+#ifdef DDS_HAS_SECURITY
```

Comment:
The conditional block here is empty because it is entering that section in the config file but ends up not generating defaults because they are automatically NULL by default. This is thus semi-intended behaviour.